atomic_reactor/plugin.py

```diff
@@ -354,6 +396,44 @@ def __init__(self, dt, workflow, plugins_conf, *args, **kwargs):
         self.plugins_results = workflow.prebuild_results
         super(PreBuildPluginsRunner, self).__init__(dt, workflow, 'PreBuildPlugin', plugins_conf, *args, **kwargs)
 
+
+class BuildStepPlugin(BuildPlugin):
+    pass
+
+
+class BuildStepPluginsRunner(BuildPluginsRunner):
+
+    def __init__(self, dt, workflow, plugin_conf, *args, **kwargs):
+        logger.info("initializing runner of build-step plugin")
+        self.plugins_results = workflow.buildstep_result
+
+        if not plugin_conf:
+            # if none buildstep plugins specified, fallback to docker api plugin
+            plugin_conf = [{'name': 'docker_api', 'is_allowed_to_fail': False}]
+        else:
```

Comment:
I don't think they're that different of use cases. For instance, if the user defines `plugin_conf` as `[{'name': 'docker_api'}]`, then it'll be modified to `[{'name': 'docker_api', 'is_allowed_to_fail': False, 'required': False}]` by the for-loop below. 

My point is that, there is at least one way that the fallback flow will result in a different plugin_conf for the same plugin. At this time, this doesn't seem to be a problem though. It just would've made the code better future proof.

I'm fine with leaving this is.